src/group/mls_group/apply_commit.rs

```diff
@@ -36,14 +36,16 @@ impl MlsGroup {
             ciphersuite,
             &commit.proposals,
             proposals_by_reference,
-            mls_plaintext.sender,
+            mls_plaintext.sender().clone(),
         ) {
             Ok(proposal_queue) => proposal_queue,
             Err(_) => return Err(ApplyCommitError::MissingProposal),
         };
 
         // Create provisional tree and apply proposals
         let mut provisional_tree = self.tree.borrow_mut();
+        // FIXME: this is a copy of the nodes in the tree to reset the original state.
```

Comment:
I know that this should be fixed with the tree rewrite. But maybe file an issue for this just to be sure?